common/djangoapps/student/views.py

```diff
@@ -1885,8 +1887,32 @@ def auto_auth(request):
     create_comments_service_user(user)
 
     # Provide the user with a valid CSRF token
-    # then return a 200 response
-    if request.META.get('HTTP_ACCEPT') == 'application/json':
+    # then return a 200 response unless redirect is true
+    if redirect_when_done:
```

Comment:
@doctoryes I've updated `redirect_when_done` to be a bool.  More specifically, it will be True if the parameter `redirect` is set to "true" (not case-sensitive).